File: src/Common/REG.cpp
#include "Common/Common.h"
#include "Common/REG.h"


std::string to_string(const REG reg) {
    switch (reg) {
        case REG::EAX: return "eax";break;
        case REG::EBP: return "ebp";break;
        case REG::EBX: return "ebx";break;
        case REG::ECX: return "ecx";break;
        case REG::EDI: return "edi";break;
        case REG::EDX: return "edx";break;
        case REG::EIP: return "eip";break;
        case REG::ESI: return "esi";break;
        case REG::ESP: return "esp";break;
        default: return "none";break;
    }
}

File: go_compiler/common/REG.py
from enum import Enum, auto


# NOTE: Start with 1
class REG(Enum):
    EAX = 0
    EBX = auto()
    ECX = auto()
    EDX = auto()
    ESI = auto()
    NONE = auto()

    EDI = auto()

    ESP = auto()
    EBP = auto()
    EIP = auto()


def to_string(reg: REG) -> str:
    return reg.name.lower()
